instrumentation-docs/src/main/java/io/opentelemetry/instrumentation/docs/DocGeneratorApplication.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.docs;
+
+import io.opentelemetry.instrumentation.docs.utils.FileManager;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class DocGeneratorApplication {
+
+  private static final Logger logger = Logger.getLogger(DocGeneratorApplication.class.getName());
+
+  public static void main(String[] args) {
+    FileManager fileManager = new FileManager("instrumentation/");
+    List<InstrumentationEntity> entities = new InstrumentationAnalyzer(fileManager).analyze();
+    printInstrumentationList(entities);
+  }
+
+  private static void printInstrumentationList(List<InstrumentationEntity> list) {
+    Map<String, List<InstrumentationEntity>> groupedByGroup =
+        list.stream()
+            .collect(
+                Collectors.groupingBy(
+                    InstrumentationEntity::getGroup, TreeMap::new, Collectors.toList()));
+
+    try (BufferedWriter writer =
+        Files.newBufferedWriter(
+            Paths.get("docs/instrumentation-list.yaml"), Charset.defaultCharset())) {
+      groupedByGroup.forEach(
+          (group, entities) -> {
+            try {
+              String groupHeader = group + ":\n  instrumentations:\n";
```

Comment:
it might be easier to use a yaml marshaller instead of writing each line manually

Reply:
updated to use snakeyml

Reply:
great 

I've done a simpler version of this for python using curcur AI - creating a python script to do the parsing: https://github.com/grafana/otel-checker/blob/bc3091b0de72b849116242c811cc303645d09b05/scripts/generate_js_supported_libraries.py

Reply:
1000 lines in java, 150 lines in python 😂

Reply:
it's not the same functionality - but python is hard to beat for this kind of task